tests/functional/conftest.py

```diff
@@ -695,3 +696,53 @@ def mock_compile(paths, base_path=None):
 
     mock.compile.side_effect = mock_compile
     return mock
+
+
+@pytest.fixture
+def mock_sepolia(ethereum, eth_tester_provider):
+    """
+    Temporarily tricks Ape into thinking the local network
+    is Sepolia so we can test features that require a live
+    network.
+    """
+    eth_tester_provider.network.name = "sepolia"
+    yield eth_tester_provider.network
+    eth_tester_provider.network.name = LOCAL_NETWORK_NAME
+
+
+@pytest.fixture
+def disable_fork_providers(ethereum):
+    """
+    When ape-hardhat or ape-foundry is installed,
+    this tricks the test into thinking they are not
+    (only uses sepolia-fork).
+    """
+    actual = ethereum.sepolia_fork.__dict__.pop("providers", {})
+    ethereum.sepolia_fork.__dict__["providers"] = {}
+    yield
+    if actual:
+        ethereum.sepolia_fork.__dict__["providers"] = actual
+
+
+@pytest.fixture
+def mock_fork_provider(mocker, ethereum):
+    """
+    A fake provider representing something like ape-foundry
+    that can fork networks (only uses sepolia-fork).
+    """
+    actual = ethereum.sepolia_fork.__dict__.pop("providers", {})
+    mock_provider = mocker.MagicMock()
+    mock_provider.name = "mock"
+    mock_provider.network = ethereum.sepolia_fork
+
+    # Have to this because providers are partials.
+    def fake_partial(*args, **kwargs):
+        mock_provider.partial_call = (args, kwargs)
+        return mock_provider
```

Comment:
question: What are partials?

Reply:
They are classes that are partially initialized, it is a functools thing iicr.. we use them for providers because we can initialize some of the provider at plugin-load time and the rest gets filled in upon connection.